services/api/mw2pdf/src/classes/MediaWikiSession.ts

```diff
@@ -1,131 +1,219 @@
+   */
+  constructor(title: string, output?: string, workDirectory?: string, makeTitlePage: boolean = false) {
+    this.title = title;
+    this.output = output;
+    this.workDirectory = workDirectory ? workDirectory : "./";
+    this.makeTitlePage = makeTitlePage;
+  }
+}
+export const MwPdfMakeTitlePage: boolean = true;
+export const MwPdfDoNoteMakeTitlePage: boolean = false;
+
 
 export class MediaWikiSession {
 
-  cookies: any;
+  cookies: Map<string, Array<ParsedCookie>>;
 
-  constructor(settings?: any) {
-    this.cookies = {}
+  constructor() {
+    this.cookies = new Map<string, Array<ParsedCookie>>();
   }
 
-  isAuthenticated(domain) { return (domain in this.cookies) }
+  isAuthenticated(domain: string): boolean {
+    return (domain in this.cookies);
+  }
 
   /**
    * Remove any stored authentication cookies.
    */
-  resetAuthentication() { this.cookies = {} }
+  resetAuthentication(): void {
+    this.cookies = new Map<string, Array<ParsedCookie>>();
+  }
 
-  getCookies(domain) { return this.cookies[domain] ?? [] }
+  getCookies(domain: string): Array<ParsedCookie> {
+    return this.cookies[domain] ?? [];
+  }
 
-  setCookies(domain, cookies) { this.cookies[domain] = cookies }
+  setCookies(domain: string, cookies: Array<ParsedCookie>): void {
+    this.cookies[domain] = cookies;
+  }
 
   /**
    * Authenticate against a given api URL.
    * If the authentication has already occurred for a given domain, this will NOT re-authenticate.
    */
-  async authenticate(username, password, apiUrl) {
-    const { host: domain } = new URL(apiUrl)
+  async authenticate(username: string, password: string, apiUrl: URL): Promise<boolean> {
+    const domain = apiUrl.host;
     if (this.isAuthenticated(domain)) {
       return true
     }
 
     // Step 1: Generate a login token
-    const loginTokenRequest = await makeLoginTokenRequest(apiUrl)
-    const authCookies = parseCookies(domain, loginTokenRequest)
-    const loginToken = await extractLoginToken(loginTokenRequest)
+    const loginTokenRequest = await makeLoginTokenRequest(apiUrl);
+    const authCookies = parseCookies(domain, loginTokenRequest);
+    const loginToken = await extractLoginToken(loginTokenRequest);
 
     // Step 2: Perform the login
-    const loginRequest = await makeLoginRequest({
+    const loginRequest = await makeLoginRequest(
       apiUrl,
       username,
       password,
       loginToken,
-      cookies: authCookies,
-    })
+      authCookies,
+    )
+
+    // Step 3: Set the cookies for the session
     this.setCookies(
       domain,
       parseCookies(domain, loginRequest),
     )
-    return true
+
+    return true;
   }
 
-  async makePdf(url, tmpDirectory = './', includeTitlePage = false) {
-    const pdfGenerator = new PdfGenerator(tmpDirectory)
+  async makePdf(url: URL, options: MwPdfOptions): Promise<Pdf> {
+    // Extract the domain (as _domain_) from the url.
```

Comment:
I'm not sure what this comment means (beyond what the code itself said already)

Reply:
Voila! Gone.